scc/light_client/retry.go

```diff
@@ -0,0 +1,115 @@
+package light_client
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+)
+
+// retryP is a provider that retries requests a certain number of times on failed
+// requests for certificates with a determined wait between retries. It is used
+// to wrap other providers to add a retry mechanism.
```

Comment:
After reading this and before looking into the implementation I was wondering: what is considered a failed request? If the request produces an error? If some requested certificates are missing? If some certificates are invalid?

I think a description of what is considered a failed request helps to understand better what can be expected from this decorator.

Reply:
well, I considering that what is a "failed" request depends on the implementation used for the internal provider. I think a more clear description would be to just clarify "retries requests that return an error"